src/lib/loaders/getCanonicalResourceDirectiveField.ts

```diff
@@ -2,31 +2,23 @@ import { FieldNode, visit, BREAK } from "graphql"
 
 export const getCanonicalResourceDirectiveForField = (
   fieldNode: FieldNode
-): string | undefined => {
-  let field: string | undefined
-  // let fullPath: string[] = []
+): string[] => {
+  const path: string[] = []
   visit(fieldNode, {
-    // SelectionSet(node, _key, _parent, _path, _ancestors) {
-    //   node.selections.map(selection => {
-    //     if (selection.kind === "Field") {
-    //       if (selection.directives && selection.directives.length) {
-    //         if (selection.directives[0].name.value === "canonicalResource") {
-    //           fullPath.push(selection.name.value)
-    //         }
-    //       }
-    //     }
-    //   })
-
-    //   return
-    // },
-    Directive(node, _key, _parent, _path, ancestors) {
+    Field: {
+      enter(node) {
+        path.push(node.name.value)
+      },
+      leave() {
+        path.pop()
+      },
```

Comment:
Keep a stack…

Reply:
Sweet, yea this makes perfect sense.

Reply:
Btw, I actually expect the error path to use field aliases when those exist, whereas here I was only using the real field name, so that needs fixing too.

Reply:
Yea, was wondering about stitching. Errors occurring using `ecommerceOrder` have that as the path (and `alias: undefined`), but the error reporting path uses `order`.

Reply:
(Will respond to the stitched errors issue outside of this conversation.)